30413/main.c

```diff
@@ -0,0 +1,245 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+MAX_LENGTH=50;
+
+typedef struct node
+{
+    int data;
+    struct node *next;
+} Node;
+Node *first;
+Node *last;
+
+void read(FILE *f,FILE *g)
+{
+    int data;
+    char *s=(char*)malloc(sizeof(char));
+    char *token=(char*)malloc(sizeof(char));
+    while(fgets(s,MAX_LENGTH,f))
+    {
+        s[strlen(s)-1]='\0';
+        token=strtok(s," ");
+
+        if(strcmp(token,"AF")==0)
+        {
+            token=strtok(NULL," ");
+            data=atoi(token);
+            addFirst(data);
+        }
+
+        if(strcmp(token,"AL")==0)
+        {
+            token=strtok(NULL," ");
+            data=atoi(token);
+            addLast(data);
+        }
+        if(strcmp(token,"DF")==0)
+        {
+            deleteFirst();
+        }
+
+        if(strcmp(token,"DL")==0)
+        {
+            deleteLast();
+        }
+        if(strcmp(token,"DOOM_THE_LIST")==0)
+        {
+            removeAll();
+        }
+        if(strcmp(token,"DE")==0)
+        {
+            token=strtok(NULL," ");
+            data=atoi(token);
+            deleteEl(data);
+        }
+        if(strcmp(token,"PRINT_ALL")==0)
+        {
+            printSLL();
+        }
+        if(strcmp(token,"PRINT_F")==0)
+        {
+            token=strtok(NULL," ");
+            data=atoi(token);
+            printFirst(data,g);
+        }
+        if(strcmp(token,"PRINT_L")==0)
+        {
+            token=strtok(NULL," ");
+            data=atoi(token);
+            printLast(data,g);
+        }
+
+    }
+}
+
+void initializeSLL()
+{
+    first=NULL;
+    last=NULL;
+}
+
+void addFirst(int data)
+{
+    if(first==NULL)
+    {
+        first=(Node*)malloc(sizeof(Node));
+        first->next=last;
+        first->data=data;
+        last=first;
+    }
+    else
+    {
+        Node *newElement=(Node*)malloc(sizeof(Node));
+        newElement->data=data;
+        newElement->next=first;
+        first=newElement;
+    }
+}
+
+void addLast(int data)
+{
+    Node *newElement=(Node*)malloc(sizeof(Node));
+    last->next=newElement;
+    newElement->next=NULL;
+    newElement->data=data;
+    last=newElement;
+}
+
+void deleteFirst()
+{
+    first=first->next;
+}
+
+
+void deleteLast()
+{
+
+    Node *currEl = first;
+    Node *prevEl = first;
+    if(first!=NULL)
+    {
+
+        while (currEl->next != NULL)
+        {
+            prevEl = currEl;
+            currEl = currEl->next;
+        }
+        prevEl->next = NULL;
+        last = prevEl;

```

Comment:
same here.. you should make sure you `free` the memory location where `last` node used to be